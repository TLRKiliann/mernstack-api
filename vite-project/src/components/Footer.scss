.div--navfooter {
  position: fixed;
  right: 0px;
  bottom: 0px;
  left: 0px;
  margin: auto;
  display: flex;
  background-color: #000000;
  border-top: 1px solid slategray;
  box-shadow: 0px 0px 10px slategray;
}

.div--navfooter ul {
  width: 100vw;
  margin: auto;
  padding-left: 0px;
  display: flex;
  justify-content: center;
}

.div--navfooter li {
  width: 20vw;
  margin: auto;
  padding: 20px 0px;
  display: flex;
  justify-content: space-around;
  font-family: 'Consolase';
  font-size: 1.4rem;
  color: lightgreen;
}