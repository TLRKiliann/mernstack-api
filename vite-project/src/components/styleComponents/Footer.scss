.div--navfooter {
  position: absolute;
  width: 100%;
  margin: auto;
  margin-bottom: auto;
  height: 80px;
  display: flex;
  justify-content: center;
  background-color: purple;
  border-bottom: 1px solid purple;
  box-shadow: 0px 0px 7px orange;

  .navbar--ul {
    width: 94vw;
    margin: auto;
    display: flex;
    justify-content: center;
  }

  .navbar--nav .navbar--ul .navbar--li {
    margin: auto;
    padding-top: 20px;
    display: flex;
    justify-content: space-around;
    font-family: "Allura", cursive;
    font-size: 2.4rem;
    font-weight: bold;
    color: whitesmoke;
    cursor: pointer;
    &:hover {
      color: orange;
    }
  }
}

@media screen and (min-width: 360px) and (max-width: 1050px) {
  .div--navfooter {
    position: absolute;
    width: 100%;
    margin: auto;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    .navbar--ul {
      width: 80vw;
      margin-left: -40px;
      display: flex;
      justify-content: center;
    }
    .navbar--nav .navbar--ul .navbar--li {
      margin: auto;
      padding-top: 10px;
      display: flex;
      justify-content: space-around;
      font-family: "Allura", cursive;
      font-size: 1.4rem;
    }
  }
}