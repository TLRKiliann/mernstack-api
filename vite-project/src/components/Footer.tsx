import { Component } from 'react'
import './styleComponents/Footer.scss'


export default class Footer extends Component {
  render() {
    return(
      <div className="div--navfooter">
        <nav className="navbar--nav">
          <ul className="navbar--ul">
            <li className="navbar--li">
              Contact
            </li>
            <li className="navbar--li">
              About US
            </li>
            <li className="navbar--li">
              Rules
            </li>
          </ul>
        </nav>
      </div>
    )
  }
}