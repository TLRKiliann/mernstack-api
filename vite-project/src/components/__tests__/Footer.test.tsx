import React from "react";
import { screen, render } from '@testing-library/react';
import { expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import Footer from "../Footer.tsx";

test('Footer get by text', () => {
  render(<Footer />);
  const linkElement = screen.getByText(/Contact/i);
  expect(linkElement).toBeInTheDocument();
});

test('MatchSnapShot test Footer', () => {
  const treeFooter = create(<Footer />)
  expect(treeFooter.toJSON()).toMatchSnapshot()
})