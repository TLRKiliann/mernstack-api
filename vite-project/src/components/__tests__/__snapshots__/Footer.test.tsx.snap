// Vitest Snapshot v1

exports[`MatchSnapShot test Footer 1`] = `
<div
  className="div--navfooter"
>
  <nav
    className="navbar--nav"
  >
    <ul
      className="navbar--ul"
    >
      <li
        className="navbar--li"
      >
        Contact
      </li>
      <li
        className="navbar--li"
      >
        Info
      </li>
      <li
        className="navbar--li"
      >
        Sponsors
      </li>
    </ul>
  </nav>
</div>
`;
